packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -1637,16 +1637,44 @@ function updatePaths(router: EmberRouter) {
   }
 
   if (!('currentPath' in appController)) {
```

Comment:
We need to wrap all the code that is set for deprecation in a deprecation flag. This allows you to build a version of Ember without any of the deprecated code. If you need an example of what you need to do I would look at. https://github.com/emberjs/ember.js/pull/17193

Reply:
I chose `export const APP_CTRL_ROUTER_PROPS = !!'3.9.0';` but that was only based on [the deprecation link](https://github.com/ember-learn/deprecation-app/pull/276/files) and my assumption that `3.9.0` refers to the version in which this is deprecated and not when the deprecation is removed.

Reply:
It's actually when it needs to be removed so it would need to be `!!'4.0.0'`.

Reply:
alright, I've updated it